src/main/java/org/commcare/util/EntitySortUtil.java

```diff
@@ -0,0 +1,83 @@
+package org.commcare.util;
+
+import org.commcare.cases.entity.Entity;
+import org.commcare.cases.util.StringUtils;
+import org.commcare.modern.util.Pair;
+import org.javarosa.core.model.instance.TreeReference;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Created by willpride on 2/21/18.
+ */
+
+public class EntitySortUtil {
+
+    public abstract static class EntitySortCallbackListener {
```

Comment:
nit but can this be called something else? `EntitySortCallbackListener` seems overly complex and non-specific when I think it would make sense to call it something like `EntityProvider`? Let me know if you disagree. Also, is there a reason this can't be an interface?